### PR TITLE
[Sui] label db handle metric with db names

### PR DESCRIPTION
**Pull request by mwtian**
## Description 

Also make sure every production DB has a name.

## Test Plan 

CI, private testnet

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-transactional-test-runner/src/simulator_persisted_store.rs [hunk 1]
```diff
@@ -97,7 +97,7 @@ impl PersistedStore {
                 self.path.clone(),
                 None,
                 None,
-                MetricConf::with_sampling(samp),
+                MetricConf::new("persisted_read").with_sampling(samp),
```
Comment (arun-koshy) [inline crates/sui-transactional-test-runner/src/simulator_persisted_store.rs:100]: readonly? same below

**Review by arun-koshy — APPROVED** [review]

File: crates/typed-store/src/rocks/mod.rs [hunk 2]
```diff
@@ -679,24 +663,25 @@ impl RocksDBBatch {
...
 pub struct MetricConf {
-    pub db_name_override: Option<String>,
+    pub db_name: String,
     pub read_sample_interval: SamplingInterval,
     pub write_sample_interval: SamplingInterval,
     pub iter_sample_interval: SamplingInterval,
 }
 
 impl MetricConf {
-    pub fn with_db_name(db_name: &str) -> Self {
```
Comment (akichidis) [inline crates/typed-store/src/rocks/mod.rs:689]: Could we maybe add an `assert` to ensure that a non empty string db name is passed?

**Review by akichidis — APPROVED** [review]
Nice! 👍

**Review by sadhansood — APPROVED** [review]
Just wanted to confirm that we do have a function called `db_name()` which uses the db path as the db name for metrics:
```
    pub fn write(self) -> Result<(), TypedStoreError> {
        let db_name = self.rocksdb.db_name();
        let _timer = self
            .db_metrics
            .op_metrics
            .rocksdb_batch_commit_latency_seconds
            .with_label_values(&[&db_name])
            .start_timer();
```
So, db metrics with db_name as a label use db path as db name

**Comment by mwtian** [comment]
> Just wanted to confirm that we do have a function called `db_name()` which uses the db path as the db name for metrics:
> 
> ```
>     pub fn write(self) -> Result<(), TypedStoreError> {
>         let db_name = self.rocksdb.db_name();
>         let _timer = self
>             .db_metrics
>             .op_metrics
>             .rocksdb_batch_commit_latency_seconds
>             .with_label_values(&[&db_name])
>             .start_timer();
> ```
> 
> So, db metrics with db_name as a label use db path as db name

Makes sense. Adding back the logic to use db path as default db name.

**Pull request merged**